### ISSUE TITLE
Back button doesn't work when live_redirect to same LiveView

### ISSUE THREAD

**Issue by pthompson** [phoenixframework/phoenix_live_view#480]
### Environment

* Elixir version (elixir -v): 1.9.1
* Phoenix version (mix deps): 1.4.11
* Phoenix LiveView version (mix deps): 0.4.1
* NodeJS version (node -v): v10.16.3
* NPM version (npm -v): 6.12.0
* Operating system: MacOS Mojave (10.14.6)

### Actual behavior

After navigating to a path which routes to the existing LiveView using live_redirect and replace: false, clicking the back button changes the url in the browser, but doesn't have any other effect.

I added a "Hello" button to the Counter example to demonstrate the issue. You can find it at https://github.com/pthompson/self_referential_push_state_example.git. 

### Expected behavior

I would expect that mount or handle_params would be called on the LiveView when the back button is clicked, allowing the LiveView to respond to the transition. Neither happens.

**Comment by chrismccord** [comment]
Can you verify if this behaviour exists with phoenix_live_view.js on master? Thanks!

**Comment by pthompson** [comment]
Yes, I updated my phoenix_live_view dependency to master and the behaviour is the same.

{:phoenix_live_view,
       git: "https://github.com/phoenixframework/phoenix_live_view.git", tag: "master"}

**Comment by aptinio** [comment]
I can also reproduce this. Using 14b70a3 on `master` also didn't help.

**Comment by feliperenan** [comment]
hi @pthompson 

I'm trying to reproduce this issue but I couldn't. See in the gif below that I'm able to get the new content after clicking on hello button and I'm also able to log the socket in the `handle_params` callback. 

![2019-11-18 17 24 21](https://user-images.githubusercontent.com/27698968/69065556-a4ea2780-0a28-11ea-9879-f4a39f11f43f.gif)

Am I missing something? 🤔

**Comment by nik-cars** [comment]
I'll see if I can throw together an independent repro this week as I am having a similar issue.

**Comment by pthompson** [comment]
@feliperenan Getting to the new content is no problem. The issue is going back. The gif you posted didn't demonstrate going back.

**Comment by nik-cars** [comment]
Oh, also, could we get a sourcemap for the liveview JS? It would make debugging easier
@chrismccord

**Comment by pthompson** [comment]
Some further information: The first back doesn't work, but if you click forward and back after that, it works as expected. I added a print to the example to show that.

**Comment by feliperenan** [comment]
I guess I've found the issue. It seems we should call `registerNewLocation` function after pushState in `onLiveRedirect`.

https://github.com/phoenixframework/phoenix_live_view/blob/14b70a3327a6c2cc24c9fab8ab56e13b2b68e131/assets/js/phoenix_live_view.js#L1134

Something like:

```JS
  onLiveRedirect({to, kind}){
    this.href = to
    Browser.pushState(kind, {}, to)
    this.liveSocket.registerNewLocation(window.location)
  }
```

At least I got it working after doing that. If maintainers agree I can send a PR :).
